Marlin/pins_RAMPS.h

```diff
@@ -162,7 +162,17 @@
   #define RAMPS_D10_PIN 10
 #endif
 
-#define HEATER_0_PIN     RAMPS_D10_PIN
+#ifdef PISTON_EXTRUDER
+  #define DRYER_FAN_PIN  RAMPS_D10_PIN //to control the dryer fan
+  #define HEATER_0_PIN   RAMPS_D9_PIN  //needed to pass sanity check
+  #ifdef E_ADJUST_MANUALLY
+    #ifndef E_ADJUST_PIN
+      #define E_ADJUST_PIN 3 //pin to read analog extruder setting
+    #endif
+  #endif
+#else
+  #define HEATER_0_PIN   RAMPS_D10_PIN
+#endif  
```

Comment:
And... Actually, when defining pins for a specific function in the pins files, don't set conditions on them. We're defining the default pins that _will be used_ when the feature that uses them is enabled. When the feature is disabled, we just don't touch the pins.